### PR TITLE
[Messenger] Adds a note InMemoryTransports will be reset in tests

### PR DESCRIPTION
**Pull request by smoench**
The `InMemoryTransport` implements the `Symfony\Contracts\Service\ResetInterface`.

### CODE REVIEW

File: messenger.rst [hunk 1]
```diff
@@ -960,6 +960,11 @@ during a request::
         }
     }
 
+.. note::
+
+        All ``in-memory`` transports will be reset automatically after each test method
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
        All ``in-memory`` transports will be resetted automatically after each test method
```

File: messenger.rst [hunk 2]
```diff
@@ -960,6 +960,11 @@ during a request::
...
+        which test class is based on the ``Symfony\Bundle\FrameworkBundle\Test\KernelTestCase``.
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
        which extends :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\KernelTestCase`.
```

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: messenger.rst [hunk 3]
```diff
@@ -960,6 +960,11 @@ during a request::
         }
     }
 
+.. note::
+
+        All ``in-memory`` transports will be reset automatically after each test **in** test classes based on the
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
        All ``in-memory`` transports will be reset automatically after each test **in**
```

File: messenger.rst [hunk 4]
```diff
@@ -960,6 +960,11 @@ during a request::
...
+        ``Symfony\Bundle\FrameworkBundle\Test\KernelTestCase`` or ``Symfony\Bundle\FrameworkBundle\Test\WebTestCase``
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
        test classes extending :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\KernelTestCase`
        or :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase`.
```

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: messenger.rst (hunk 3 above)
Comment (smoench) [reply messenger.rst]: Done

File: messenger.rst (hunk 4 above)
Comment (smoench) [reply messenger.rst]: Done

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Simon.

**Comment by smoench** [comment]
Thank you, @OskarStark!

**Pull request merged**